ast.c

```diff
@@ -724,6 +720,53 @@ static void ast_to_zval(zval *zv, zend_ast *ast, ast_state_info_t *state) {
 			ast->kind = ZEND_AST_ASSIGN_OP;
 			ast->attr = AST_BINARY_COALESCE;
 			break;
+		case ZEND_AST_CLASS_NAME:
+			if (state->version < 70) {
+				zval name_zval;
+				ast_to_zval(&name_zval, ast->child[0], state);
+				zval class_name_zval;
+				ast_create_virtual_node_ex(
+						&class_name_zval, AST_NAME, ast->child[0]->attr, zend_ast_get_lineno(ast), state, 1, &name_zval);
+				zval const_zval;
+				ZVAL_STR_COPY(&const_zval, AST_STR(str_class));
+				ast_create_virtual_node_ex(
+						zv, ZEND_AST_CLASS_CONST, 0, zend_ast_get_lineno(ast), state, 2, &class_name_zval, &const_zval);
+				return;
+			}
+			break;
+#else
+		case ZEND_AST_CLASS_CONST:
+			if (state->version >= 70) {
+				// Convert to an AST_CLASS_NAME instead. This is the opposite of the work done in the ZEND_AST_CLASS_NAME case.
+				zend_ast *const_name_ast = ast->child[1];
+				if (const_name_ast->kind != ZEND_AST_ZVAL) {
```

Comment:
While the class can be an expression, the constant name can't be (we don't have `Foo::{$bar}` in PHP), so these checks at least can be replaced with `zend_ast_get_str`.